### PR TITLE
bump tcgc and typespec-java, support `api-version`

### PR DESCRIPTION
**Pull request by XiaofeiCao**
fix https://github.com/Azure/autorest.java/issues/2720
1. support `api-version` in tspconfig.yaml
2. bump tcgc to 0.41.6
3. bump typespec-java to 0.15.13 for release

### CODE REVIEW

File: typespec-extension/src/code-model-builder.ts [hunk 1]
```diff
@@ -181,6 +181,7 @@ import {
 } from "./operation-utils.js";
 import { isArmCommonType } from "./type-utils.js";
 import pkg from "lodash";
+import { log } from "console";
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:184]: log not used?

File: typespec-extension/src/code-model-builder.ts [hunk 2]
```diff
@@ -213,8 +214,8 @@ export class CodeModelBuilder {
     this.sdkContext = createSdkContext(context, "@azure-tools/typespec-java");
     if (this.supportsAdvancedVersioning()) {
       this.sdkContext.apiVersion = "all";
-    } else {
-      this.sdkContext.apiVersion = "latest";
+    } else if (this.options["api-version"]) {
+      this.sdkContext.apiVersion = this.options["api-version"];
```
Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:218]: Could it be "latest" or "all" here?

Comment (weidongxu-microsoft) [inline typespec-extension/src/code-model-builder.ts:218]: Maybe we don't set this at all.

Instead, we add another `api-version=all` to the cases we now have `advancedVersioning`.

File: typespec-tests/Generate.ps1 [hunk 3]
```diff
@@ -43,13 +43,18 @@ $generateScript = {
   } elseif ($tspFile -match "resiliency[\\/]srv-driven[\\/]main.tsp") {
     # enable advanced versioning for resiliency test
     $tspOptions += " --option ""@azure-tools/typespec-java.advanced-versioning=true"""
```
Comment (weidongxu-microsoft) [inline typespec-tests/Generate.ps1:45]: For comment https://github.com/Azure/autorest.java/pull/2722#discussion_r1580502052

We could just add `api-version=all` to this case.

File: typespec-extension/src/code-model-builder.ts (hunk 2 above)
Comment (XiaofeiCao) [reply typespec-extension/src/code-model-builder.ts]: Yeah, it'll read from emitter-options. Removed.

File: typespec-extension/src/code-model-builder.ts (hunk 1 above)
Comment (XiaofeiCao) [reply typespec-extension/src/code-model-builder.ts]: No, removed. It's added accidentally by using IDE...

File: typespec-tests/Generate.ps1 (hunk 3 above)
Comment (XiaofeiCao) [reply typespec-tests/Generate.ps1:46]: Added.

File: typespec-extension/src/code-model-builder.ts (hunk 2 above)
Comment (XiaofeiCao) [reply typespec-extension/src/code-model-builder.ts]: Updated.

**Review by weidongxu-microsoft — APPROVED** [review]

**Pull request merged**